File: java/two-fer/Twofer.java
/**
 * Two Fer
 * https://exercism.org/tracks/java/exercises/two-fer
 */
public class Twofer {

    private static final String MESSAGE = "One for %s, one for me.";

    public String twofer(String name) {
        if (name != null) {
            return String.format(MESSAGE, name);
        } else {
            return String.format(MESSAGE, "you");
        }
    }
}

File: kotlin/TwoFer.kt
/**
 * https://exercism.io/my/tracks/kotlin
 * Two Fer
 */

fun twofer_1(name: String = "you"): String {
    return "One for $name, one for me."
}

fun twofer_2(name: String = "you") = "One for $name, one for me."

fun twofer_3(name: String = "you"): String {
    return listOf("One for $name", "one for me").joinToString(separator = ", ", postfix = ".")
}
